src/internal/components/option/utils/unflatten-options.ts

```diff
@@ -27,7 +27,7 @@ export function unflattenOptions(options: ReadonlyArray<DropdownOption>): Nested
       const wrapped: ParentDropdownOption = { type: 'parent', option, index, children: [] };
       currentParent = wrapped;
       nestedOptions.push(wrapped);
-    } else if (!option.type || option.type === 'child') {
+    } else if (option.type === 'child') {
```

Comment:
What is the case for the `option.type` to be undefined?

Reply:
Assuming that root-level non-parent nodes have `option.type` set to `undefined`, I think this can be on the right direction, and simpler than [my approach](https://github.com/cloudscape-design/components/pull/4037/files).

Could you add some testing coverage?